### ISSUE TITLE
Speed up execution suggestion.

### ISSUE THREAD

**Issue by a904guy** [iam-abbas/Reddit-Stock-Trends#3]
Cache the calls to Yahoo Finance to speed up execution.

You should replace all the calls to `yf.Ticker(symbol)` with this function:

```
_symbols = {}
def yfTicker(symbol):
  if(symbol in _symbols):
    return _symbols[symbol]
  _symbols[symbol] = yf.Ticker(symbol)
  return _symbols[symbol]
```

Additionally when using the notebook you can just restart the kernel or run `_symbols = {}` again to clear the caching.

### Why?

Because every call is being executed when yf.Ticker(symbol) is called, it's a ton of redundant calls to YF especially when your calling the same ticker repeatedly.

Additional possible speedup would to overload the history functions in a similar manner as above as well.

**Comment by iam-abbas** [comment]
That's a great idea @a904guy! In addition to this, I am also going to add a local static JSON/CSV containing all the tickers to verify the Tickers because api calls could be time taking. Suggestion by @Denbergvanthijs

**Comment by a904guy** [comment]
Noticed I had a bug in the function, was calling `tic` not `symbol`. I've updated the example

**Comment by a904guy** [comment]
History Caching to speed up Yahoo Financial section.

```_symbols_history = {}
def yfTickerHistory(symbol, period = '1d'):
  if(symbol in _symbols_history and period in _symbols_history[symbol]):
     return _symbols_history[symbol][period]
  if(symbol not in _symbols_history):
    _symbols_history[symbol] = {}
  _symbols_history[symbol][period] = yfTicker(symbol).history(period)
  return _symbols_history[symbol][period]```

**Comment by iam-abbas** [comment]
Fixed in #12
